src/io/mod.rs

```diff
@@ -152,6 +153,80 @@ impl Io {
     }
 }
 
+/// A struct that wraps a csv `Reader` and provides methods for reading one record at a time.
+/// It also implements `Iterator`.
+pub struct DelimFileReader<D: DeserializeOwned> {
+    record_iter: DeserializeRecordsIntoIter<Box<dyn BufRead + Send>, D>,
+}
+
+impl<D: DeserializeOwned> DelimFileReader<D> {
+    /// Returns a new `DelimFileReader` that will read records from the given reader with the given
+    /// delimiter and quoting. Assumes the input file has a header row.
+    pub fn new(reader: Box<dyn BufRead + Send>, delimiter: u8, quote: bool) -> Self {
+        let csv_reader = ReaderBuilder::new()
+            .delimiter(delimiter)
+            .has_headers(true)
+            .quoting(quote)
+            .from_reader(reader);
+        let record_iter = csv_reader.into_deserialize();
+        Self { record_iter }
```

Comment:
shall we also store a copy of the header?  That may prove useful later, or we can add it in later  https://docs.rs/csv/1.3.0/csv/struct.Reader.html#method.headers

Reply:
Ah good point. Have made that change.